parallel-install/pkg/installation/installation_test.go

```diff
@@ -0,0 +1,352 @@
+
+const cancelTimeout = 150 * time.Millisecond
+const quitTimeout = 250 * time.Millisecond
+
+func TestInstallation_StartKymaInstallation(t *testing.T) {
+	t.Parallel()
+
+	i := newInstallation()
+
+	t.Run("should install Kyma", func(t *testing.T) {
+		kubeClient := fake.NewSimpleClientset()
+
+		hc := &mockHelmClient{}
+		provider := &mockProvider{
+			hc: hc,
+		}
+		overridesProvider := &mockOverridesProvider{}
+		eng := engine.NewEngine(overridesProvider, provider, "", engine.Config{
+			WorkersCount: 2,
+			Log:          log.Printf,
+		})
+
+		err := i.StartKymaInstallation(kubeClient, provider, overridesProvider, eng)
+
+		assert.NoError(t, err)
+	})
+
+	t.Run("should fail to install Kyma prerequisites", func(t *testing.T) {
+		t.Run("due to cancel timeout", func(t *testing.T) {
+			kubeClient := fake.NewSimpleClientset()
+
+			hc := &mockHelmClient{
+				componentProcessingTime: 200,
+			}
+			provider := &mockProvider{
+				hc: hc,
+			}
+			overridesProvider := &mockOverridesProvider{}
+			eng := engine.NewEngine(overridesProvider, provider, "", engine.Config{
+				WorkersCount: 2,
+				Log:          log.Printf,
+			})
+
+			err := i.StartKymaInstallation(kubeClient, provider, overridesProvider, eng)
+
+			assert.Error(t, err)
+			assert.EqualError(t, err, "Kyma prerequisites installation failed due to the timeout")
+		})
+		t.Run("due to quit timeout", func(t *testing.T) {
+			kubeClient := fake.NewSimpleClientset()
+
+			hc := &mockHelmClient{
+				componentProcessingTime: 300,
+			}
+			provider := &mockProvider{
+				hc: hc,
+			}
+			overridesProvider := &mockOverridesProvider{}
+			eng := engine.NewEngine(overridesProvider, provider, "", engine.Config{
+				WorkersCount: 2,
+				Log:          log.Printf,
+			})
+
+			start := time.Now()
+			err := i.StartKymaInstallation(kubeClient, provider, overridesProvider, eng)
+			end := time.Now()
+
+			elapsed := end.Sub(start)
+
+			assert.Error(t, err)
+			assert.EqualError(t, err, "Force quit: Kyma prerequisites installation failed due to the timeout")
+
+			// One component installation lasts 300 ms
+			// Quit timeout occurs at 250 ms
+			// Check if program quits in the meantime
+			assert.GreaterOrEqual(t, elapsed.Milliseconds(), int64(250))
+			assert.Less(t, elapsed.Milliseconds(), int64(300))
+		})
+	})
+
+	t.Run("should install prerequisites and fail to install Kyma components", func(t *testing.T) {
+		t.Run("due to cancel timeout", func(t *testing.T) {
+			kubeClient := fake.NewSimpleClientset()
+
+			hc := &mockHelmClient{
+				componentProcessingTime: 40,
+			}
+			provider := &mockProvider{
+				hc: hc,
+			}
+			overridesProvider := &mockOverridesProvider{}
+			eng := engine.NewEngine(overridesProvider, provider, "", engine.Config{
+				WorkersCount: 2,
+				Log:          log.Printf,
+			})
+
+			err := i.StartKymaInstallation(kubeClient, provider, overridesProvider, eng)
+
+			assert.Error(t, err)
+			assert.EqualError(t, err, "Kyma installation failed due to the timeout")
```

Comment:
Please also add time-based assertions with a comment as in "due to quit timeout" test case